bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/InverterHandler.java

```diff
@@ -103,12 +103,52 @@ protected void handlePolledData(ModbusRegisterArray registers) {
                                                                                   // see:
                                                                                   // https://github.com/openhab/openhab-core/pull/1347
 
+        // DC General group
+        updateState(new ChannelUID(getThing().getUID(), GROUP_DC_GENERAL, CHANNEL_DC_CURRENT),
+                getScaled(block.dcCurrent, block.dcCurrentSF, AMPERE));
+        updateState(new ChannelUID(getThing().getUID(), GROUP_DC_GENERAL, CHANNEL_DC_VOLTAGE),
+                getScaled(block.dcVoltage, block.dcVoltageSF, VOLT));
+        updateState(new ChannelUID(getThing().getUID(), GROUP_DC_GENERAL, CHANNEL_DC_POWER),
+                getScaled(block.dcPower, block.dcPowerSF, VOLT));
+
         updateState(channelUID(GROUP_AC_GENERAL, CHANNEL_AC_POWER_FACTOR),
                 getScaled(block.acPowerFactor, block.acPowerFactorSF, PERCENT));
 
         updateState(channelUID(GROUP_AC_GENERAL, CHANNEL_AC_LIFETIME_ENERGY),
                 getScaled(block.acEnergyLifetime, block.acEnergyLifetimeSF, WATT_HOUR));
 
+        // AC Phase specific groups
+        // All types of inverters
+        updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_A, CHANNEL_AC_PHASE_CURRENT),
+                getScaled(block.acCurrentPhaseA, block.acCurrentSF, AMPERE));
+        updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_A, CHANNEL_AC_VOLTAGE_TO_NEXT),
+                getScaled(block.acVoltageAB, block.acVoltageSF, VOLT));
+        updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_A, CHANNEL_AC_VOLTAGE_TO_N),
+                getScaled(block.acVoltageAtoN, block.acVoltageSF, VOLT));
+
+        // Split phase and three phase
+        if ((thing.getThingTypeUID().equals(THING_TYPE_INVERTER_SPLIT_PHASE)
+                || thing.getThingTypeUID().equals(THING_TYPE_INVERTER_THREE_PHASE))
+                && block.phaseConfiguration >= INVERTER_SPLIT_PHASE) {
+            updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_B, CHANNEL_AC_PHASE_CURRENT),
+                    getScaled(block.acCurrentPhaseB, block.acCurrentSF, AMPERE));
+            updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_B, CHANNEL_AC_VOLTAGE_TO_NEXT),
+                    getScaled(block.acVoltageBC, block.acVoltageSF, VOLT));
+            updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_B, CHANNEL_AC_VOLTAGE_TO_N),
+                    getScaled(block.acVoltageBtoN, block.acVoltageSF, VOLT));
+        }
+
+        // Three phase only
+        if (thing.getThingTypeUID().equals(THING_TYPE_INVERTER_THREE_PHASE)
+                && block.phaseConfiguration >= INVERTER_THREE_PHASE) {
+            updateState(new ChannelUID(getThing().getUID(), GROUP_AC_PHASE_C, CHANNEL_AC_PHASE_CURRENT),
```

Comment:
These should be using your `channelUID(...)` method in instead.

Reply:
Thanks, I've overlooked this. Fixed